primitives/src/circuit/schnorr_dsa.rs

```diff
@@ -64,7 +64,7 @@ where
     /// * `msg` - message variables that have been signed.
     /// * `sig` - signature variable.
     /// * `returns` - a bool variable indicating whether the signature is valid.
-    fn is_valid_signature(
+    fn check_signature_validity(
```

Comment:
I think the (implicit) naming convention in current code is:
for gadgets that "enforces" a relation among input params, we use `check_xxx() -> Result<(), Error>`; 
for gadgets that produces a boolean bit about the SAT of relation among the input params, we use `is_xxx() -> Result<Variable, Error>`

Here maybe we should stick with the existing name?

Reply:
yeah. clippy complaint about the use of `is_xxx()` -- such a function should not mutate `self` which makes sense. Do we have other options than `is_xxx()`?

Reply:
Maybe: change all current `is_xxx()` (such as [here](https://github.com/EspressoSystems/jellyfish/blob/main/plonk/src/circuit/customized/mod.rs#L326)) to `check_xxx()` and those that enforce and will return error if not SAT, we keep using `xxx_gate()` (like [this](https://github.com/EspressoSystems/jellyfish/blob/main/plonk/src/circuit/customized/mod.rs#L316))?
as long as we are consistent, I'm fine.